### PR TITLE
Kotlinize

### PR DESCRIPTION
**Pull request by hannesa2**
During https://github.com/owncloud/android/pull/2679 I kotlinize some files and fixed some lint warnings. 
To keep https://github.com/owncloud/android/pull/2679 I separated my work into two PR.

### CODE REVIEW

**Comment by davigonz** [comment]
Wow, this is great, thanks so much!

File: owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,743 @@
...
+                            parentActivity.resources,
+                            thumbnail,
+                            task
+                        )
+                        fileIcon.setImageDrawable(asyncDrawable)
+                        task.execute(fakeFileToCheatThumbnailsCacheManagerInterface)
+                    }
+                }
+
+                if ("image/png" == upload.mimeType) {
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt]: "image/png" could be included as a val

File: owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,743 @@
...
+                        ContextCompat.getColor(parentActivity, R.color.background_color)
+                    )
+                }
+
+            } else if (fakeFileToCheatThumbnailsCacheManagerInterface.isImage) {
+                val file = File(upload.localPath)
+                // Thumbnail in Cache?
+                var thumbnail = ThumbnailsCacheManager.getBitmapFromDiskCache(
+                    file.hashCode().toString()
+                )
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt]: Same here

File: owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,743 @@
...
+                        fileIcon.setImageDrawable(asyncDrawable)
+                        task.execute(file)
+                        Log_OC.v(TAG, "Executing task to generate a new thumbnail")
+                    }
+                }
+
+                if ("image/png".equals(upload.mimeType, ignoreCase = true)) {
+                    fileIcon.setBackgroundColor(
+                        ContextCompat.getColor(parentActivity, R.color.background_color)
+                    )
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt]: Same here

File: owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,743 @@
...
+                }
+            } else {
+                fileIcon.setImageResource(
+                    MimetypeIconUtil.getFileTypeIconId(
+                        upload.mimeType,
+                        fileName
+                    )
+                )
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt]: Same here

File: owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,743 @@
...
+     * Gets the status text to show to the user according to the status and last result of the
+     * the given upload.
+     *
+     * @param upload        Upload to describe.
+     * @return Text describing the status of the given upload.
+     */
+    private fun getStatusText(upload: OCUpload): String {
+
+        var status: String
+        when (upload.uploadStatus) {
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt]: Do we need that status variable? Couldn't it be reformat to `return when() { ... }`

File: owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,743 @@
...
+        loadUploadItemsFromDb()
+    }
+
+    override fun getChild(groupPosition: Int, childPosition: Int): Any {
+        return uploadGroups[getGroupId(groupPosition).toInt()].items[childPosition]
+    }
+
+    override fun getChildId(groupPosition: Int, childPosition: Int): Long {
+        return childPosition.toLong()
+    }
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt]: single line function better

File: owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt [hunk 7]
```diff
@@ -0,0 +1,743 @@
...
+            uploadGroups[getGroupId(groupPosition).toInt()].items,
+            childPosition,
+            convertView,
+            parent
+        )
+    }
+
+    override fun getChildrenCount(groupPosition: Int): Int {
+        return uploadGroups[getGroupId(groupPosition).toInt()].items.size
+    }
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt]: Single line function better

File: owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,743 @@
...
+
+    override fun getGroup(groupPosition: Int): Any {
+        return uploadGroups[getGroupId(groupPosition).toInt()]
+    }
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt]: Single line function better

File: owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt [hunk 9]
```diff
@@ -0,0 +1,743 @@
...
+
+    override fun getGroupCount(): Int {
+        var size = 0
+        for (uploadGroup in uploadGroups) {
+            if (uploadGroup.items.size > 0) {
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt]: isNotEmpty() better?

File: owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt [hunk 10]
```diff
@@ -0,0 +1,743 @@
...
+        var convertViewLocal = convertView
+        //force group to stay unfolded
+        val listView = parent as ExpandableListView
+        listView.expandGroup(groupPosition)
+
+        listView.setGroupIndicator(null)
+        val group = getGroup(groupPosition) as UploadGroup
+        if (convertView == null) {
+            val inflater =
+                parentActivity.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt]: Same line better

File: owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt [hunk 11]
```diff
@@ -0,0 +1,743 @@
...
+            String.format(parentActivity.getString(stringResFileCount), group.groupCount)
+
+        tvGroupName?.text = group.groupName
+        tvFileCount?.text = fileCountText
+        return convertViewLocal!!
+    }
+
+    override fun isChildSelectable(groupPosition: Int, childPosition: Int): Boolean {
+        return true
+    }
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt]: Single line function

File: owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt [hunk 12]
```diff
@@ -0,0 +1,743 @@
...
+        }
+
+        override fun onTransferProgress(
+            progressRate: Long,
+            totalTransferredSoFar: Long,
+            totalToTransfer: Long,
+            filename: String
+        ) {
+            val percent =
+                (100.0 * totalTransferredSoFar.toDouble() / totalToTransfer.toDouble()).toInt()
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt]: Same line

File: owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt [hunk 13]
```diff
@@ -0,0 +1,743 @@
...
+    override fun isChildSelectable(groupPosition: Int, childPosition: Int): Boolean {
+        return true
+    }
+
+    inner class ProgressListener(upload: OCUpload, progressBar: ProgressBar) :
+        OnDatatransferProgressListener {
+        private var lastPercent = 0
+        var upload: OCUpload? = null
+            internal set
+        private var weakProgressBar: WeakReference<ProgressBar>? = null
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt]: Maybe if you set this as not nullable, you won't need to assert it later (?)

File: owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt [hunk 14]
```diff
@@ -0,0 +1,743 @@
...
+        }
+        val tvGroupName = convertViewLocal?.findViewById<TextView>(R.id.uploadListGroupName)
+        val tvFileCount = convertViewLocal?.findViewById<TextView>(R.id.textViewFileCount)
+
+        val stringResFileCount = if (group.groupCount == 1)
+            R.string.uploads_view_group_file_count_single
+        else
+            R.string.uploads_view_group_file_count
+        val fileCountText =
+            String.format(parentActivity.getString(stringResFileCount), group.groupCount)
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt]: Same line

File: owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt [hunk 15]
```diff
@@ -0,0 +1,743 @@
...
+                    )
+                UploadResult.UPLOADED ->
+                    // should not get here ; status should be UPLOAD_SUCCESS
+                    status =
+                        parentActivity.getString(R.string.uploads_view_upload_status_succeeded)
+                UploadResult.SPECIFIC_FORBIDDEN ->
+                    // We don't know the specific forbidden error message because it is not being
+                    // saved in uploads storage
+                    status =
+                        String.format(parentActivity.getString(R.string.uploader_upload_forbidden))
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt]: Do you need a String format here?

File: owncloudApp/src/main/java/com/owncloud/android/broadcastreceivers/ConnectivityActionReceiver.kt [hunk 16]
```diff
@@ -0,0 +1,153 @@
...
+         *
+         * But first let's try something "simple" to keep a basic retry of camera uploads in
+         * version 1.9.2, similar to the existent until 1.9.1. To be improved.
+         */
+        if (intent.action == WifiManager.NETWORK_STATE_CHANGED_ACTION) {
+            val networkInfo = intent.getParcelableExtra<NetworkInfo>(WifiManager.EXTRA_NETWORK_INFO)
+            val wifiInfo = intent.getParcelableExtra<WifiInfo>(WifiManager.EXTRA_WIFI_INFO)
+            val bssid = intent.getStringExtra(WifiManager.EXTRA_BSSID)
+            if (networkInfo.isConnected &&      // not enough; see (*) right below
+
```
[the remark this answers is not in the document]
Comment (hannesa2) [reply owncloudApp/src/main/java/com/owncloud/android/broadcastreceivers/ConnectivityActionReceiver.kt]: Please consider to use ktLint, then you can make a lot of such checks. 
But with your Bitrise, I've no clue how to include this build step

**Comment by hannesa2** [comment]
I applied most of the suggestions

**Comment by abelgardep** [comment]
Thanks for the changes! But there are some pending ones that require some attention and research. As you know, we may get some nullability problems if we have kotlin code running with java code with not nullable parameters. Some files in this pr have a lot of usages and we should research a little bit which params are nullable and which not. That's why is not trivial to kotlinize files.

For example, i have found an usage of uploadNewFile from UriUploader.java that sends a null to a parameter set as not nullable.

File: owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt [hunk 17]
```diff
@@ -0,0 +1,743 @@
...
+            }
+
+            /// Set icon or thumbnail
+            val fileIcon = view.findViewById<ImageView>(R.id.thumbnail)
+            fileIcon.setImageResource(R.drawable.file)
+
+            /** Cancellation needs do be checked and done before changing the drawable in fileIcon, or
+             * [ThumbnailsCacheManager.cancelPotentialWork] will NEVER cancel any task.
+             */
+            @Suppress("KDocUnresolvedReference")
```
[the remark this answers is not in the document]
Comment (hannesa2) [reply owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt]: I've no clue how, you are warmly welcome to add a commit, when you know how

File: owncloudApp/src/main/java/com/owncloud/android/broadcastreceivers/ConnectivityActionReceiver.kt (hunk 16 above)
[the remark this answers is not in the document]
Comment (michaelstingl) [reply owncloudApp/src/main/java/com/owncloud/android/broadcastreceivers/ConnectivityActionReceiver.kt]: > But with your Bitrise, I've no clue how to include this build step

Looks like Bitrise supports ktLint:
https://blog.bitrise.io/tags/ktlint

**Comment by hannesa2** [comment]
Sorry, I missed them because of this
image

But wouldn't is better and **more efficient** when you change immediate in code and commit your code-format wishes ? Tastes are different and efficiency matters !

Most of them are autogenerated by Android Studio

**Comment by abelgardep** [comment]
Autogenerate kotlin code from java using AS is not always the best solution. If i kotlinize a file, i should research and check if everything is working as it was before since we have not a full test coverage (we are on our way with new arch). I found that problem but perhaps there are even more.

**Comment by jesmrec** [comment]
@hannesa2 do you mind if we kotlinize these classes in the re-architecting process, or do you prefer to apply the suggestions of @abelgardep to move forward this PR?

**Comment by hannesa2** [comment]
I've no strong opinion here, I don't depend on it. Tell me what to do. Even an unmerged "close" is an option

**Comment by jesmrec** [comment]
Ok. Let's close ftm. In case we want to take advantage of this, will reopen. 

Thanks for your enagagement.

**Pull request closed without merging**